src/Microsoft.Health.Fhir.Shared.Api/Controllers/FhirController.cs

```diff
@@ -368,6 +369,26 @@ public async Task<IActionResult> Delete(string typeParameter, string idParameter
             return FhirResult.NoContent().SetETagHeader(response.WeakETag);
         }
 
+        /// <summary>
+        /// Deletes the specified resource
+        /// </summary>
+        /// <param name="typeParameter">The type.</param>
+        /// <param name="hardDelete">A flag indicating whether to hard-delete the resource or not.</param>
+        /// <param name="multiple">A flag indicating whether to delete all resources that are found in the search.</param>
+        [HttpDelete]
+        [Route(KnownRoutes.ResourceType)]
+        [AuditEventType(AuditEventSubType.Delete)]
+        public async Task<IActionResult> ConditionalDelete(string typeParameter, [FromQuery] bool hardDelete, [FromQuery] bool multiple)
+        {
+            IReadOnlyList<Tuple<string, string>> conditionalParameters = GetQueriesForSearch();
+
+            DeleteResourceResponse response = await _mediator.Send(new ConditionalDeleteResourceRequest(typeParameter, conditionalParameters, hardDelete, multiple), HttpContext.RequestAborted);
+
+            Response.Headers.Add(KnownHeaders.ItemsDeleted, (response?.ResourcesDeleted ?? 0).ToString(CultureInfo.InvariantCulture));
```

Comment:
Is this useful? Or should it not expose this value?

Reply:
Wrap it in `if (multiple)` ?

Reply:
Makes sense